src/components/Settings/composables/useNavigation.ts

```diff
@@ -66,8 +74,27 @@ export const navigations: {
 const useSettingsNavigation = () => {
   const router = useRouter()
   const { lastOpenChannelName } = useBrowserSettings()
-  const close = () =>
-    router.push(constructChannelPath(lastOpenChannelName.value))
+  const { channelIdToShortPathString } = useChannelPath()
+
+  const close = () => {
+    switch (primaryView.value.type) {
+      case 'channel':
+        router.push(
+          constructChannelPath(
+            channelIdToShortPathString(primaryView.value.channelId)
+          )
+        )
+        break
+      case 'clips':
+        router.push(constructClipFoldersPath(primaryView.value.clipFolderId))
+        break
+      case 'dm':
+        router.push(constructUserPath(primaryView.value.userName))
+        break
+      default:
+        router.push(constructChannelPath(lastOpenChannelName.value))
```

Comment:
```suggestion
      case 'qall':
        // TODO(sapphi-red): PrimaryViewにqallが来ることは想定していないが、とりあえず書く
        router.push(constructChannelPath(lastOpenChannelName.value))
        break
      default:
        const check: never = primaryView.value.type
        throw new Error(`Unknown view type: ${check}`)
```
こんな感じに[exhaustive check](https://typescript-jp.gitbook.io/deep-dive/type-system/discriminated-unions#chekkuexhaustive-checks)をしておくとよかったりします 😉

(caseが`qall`のときは本来来ない設計なのに型に残っていたので、あとで自分が消しておきます:pray: とりあえずは書いた通りにしておいてもらえると…)